File: Leetcode/287-Find_the_Duplicate_Number.java
import java.util.*;
class Solution {
    public int findDuplicate(int[] nums) {
        Arrays.sort(nums);
        int ans=0;
        for(int i=0 ; i<nums.length-1 ; i++){
            if(nums[i]==nums[i+1])ans= nums[i];
        }
        return ans;
    }
}

File: Leetcode-Solutions/287-Find_the_Duplicate_Number.cpp
#include <vector>
using namespace std;
class Solution {
public:
  int findDuplicate(vector<int> &nums) {
    vector<bool> flag(nums.size(), false);
    int ans = -1;
    for (int i = 0; i < nums.size(); i++) {
      if (flag[nums[i]]) {
        ans = nums[i];
        break;
      }
      flag[nums[i]] = true;
    }
    return ans;
  }
};
